Automattic/woocommerce-payments
# Dev: Adjust E2E tests to work locally on Apple silicon chips, and improve some tests

#### Changes proposed in this Pull Request

This PR includes the image update from `wordpress:5.6.2-php7.3` to `wordpress:6.0.2-php7.4` which is a most recent version and works better on Docker environments running on Apple Silicon with virtiofs enabled. Also changes the image user from `xfs` to `www-root` which seems to be the predefined user in the image. Source: https://hub.docker.com/_/wordpress#running-as-an-arbitrary-user

#### Testing instructions

* All tests should pass on
    * GH Actions
    * Environments on Intel chips
    * Environments on Apple Silicon not using virtiofs 
    * Environments on Apple Silicon using virtiofs

Steps to test:
1. Get the latest code from this branch using `git checkout update/e2e-test-image`
2. Run `npm install && composer install && npm run build:client` to update the dependencies and builds
4. Run `npm run test:e2e-reset && npm run test:e2e-setup` to set the E2E test environment up
5. Run `npm run test:e2e` to start the testing.

> **Warning**: You may need to add `WP_BASE_URL=http://localhost:8084` to your `local.env` file if you're getting HTTP 400 errors to calls made to `http://localhost:8889`. In the latest version of `woocommerce/e2e-environment` package, this seems to be hardcoded on its dependencies as the testing container URL, and you may need to override that.

QA Testing Not Applicable

-------------------


**Post merge**
